django/contrib/auth/base_user.py

```diff
@@ -138,4 +138,4 @@ def get_email_field_name(cls):
 
     @classmethod
     def normalize_username(cls, username):
-        return unicodedata.normalize('NFKC', username) if username else username
+        return unicodedata.normalize('NFKC', username) if type(username) is str else username
```

Comment:
Seems okay, although I wonder if `isinstance(username, str)` would be fine? A test is also required. Possibly the problem could be reproduced with `tests.auth_tests.models.IntegerUsernameUser`.